src/bp/ui-studio/src/web/views/OneFlow/diagram/TriggerEditor/SingleParam.tsx

```diff
@@ -1,59 +1,105 @@
-import { Checkbox, FormGroup, InputGroup, NumericInput } from '@blueprintjs/core'
-import { ConditionListOptions } from 'botpress/sdk'
+import { Checkbox, FormGroup, InputGroup, NumericInput, Radio, RadioGroup, Switch, TextArea } from '@blueprintjs/core'
+import { ConditionParam } from 'botpress/sdk'
+import { lang } from 'botpress/shared'
 import _ from 'lodash'
 import React, { FC } from 'react'
 import ContentPickerWidget from '~/components/Content/Select/Widget'
 
 import Dropdown from './Condition/Dropdown'
 
-interface Props {
-  label: string
+type Props = {
   value?: any
-  type: 'string' | 'number' | 'boolean' | 'list' | 'content'
-  list?: ConditionListOptions
   updateValue: (value) => void
-}
+} & ConditionParam
 
 const SingleParam: FC<Props> = props => {
-  const { value, label, type, list, updateValue } = props
-
-  if (type === 'string') {
-    return (
-      <FormGroup key={label} label={label}>
-        <InputGroup value={value || ''} onChange={e => updateValue(e.currentTarget.value)} />
-      </FormGroup>
-    )
-  } else if (type === 'number') {
-    return (
-      <FormGroup key={label} label={label}>
-        <NumericInput value={value || 0} onValueChange={value => updateValue(value)} />
-      </FormGroup>
-    )
-  } else if (type === 'list') {
-    return (
-      <FormGroup key={label} label={label}>
-        <Dropdown
-          listOptions={list}
-          defaultValue={value}
-          selectedValue={value}
-          onChange={item => updateValue(item && item.value)}
-        />
-      </FormGroup>
-    )
-  } else if (type === 'boolean') {
-    return <Checkbox key={label} label={label} checked={value} onChange={e => updateValue(e.currentTarget.checked)} />
-  } else if (type === 'content') {
-    return (
-      <FormGroup key={label} label={label}>
-        <ContentPickerWidget
-          style={{ zIndex: 0 }}
-          name="contentPicker"
-          id="contentPicker"
-          itemId={value}
-          onChange={item => updateValue(item.id)}
-        />
-      </FormGroup>
-    )
+  const { value, label, type, subType, list, required, updateValue } = props
```

Comment:
3 suggestions regarding this as it could become harder to maintain with more complex conditions. For now this is fine but when we'll try to incorporate a web designer's input for this, I expect it to become really complex.

1-  we use json schema + json schema renderer ...
2- we make this as extensible as possible and use this for content as well ( or maybe it'll go away ? )
3- we enforce the module developer to develop a UI component for each trigger, making this super generic code obsolete.

Let me know what you think

Reply:
Not enough flexibility with json schema renderer, it would require a lot of custom components to be developed, which would take too much time for now, but could be an option for the future

We definitely want to avoid 3) since it's the mistake we did with skills, with the result that nobody creates any of them because they are complicated. 

So we should go with 2). Basically, if we can define everything in an object, then we can also let the user build his "form" this way for packaged skills